### DISCUSSION TITLE
Security advantage of private-tmp (in firefox.profile)

### DISCUSSION THREAD

**Discussion by BreakfastSerial** [netblue30/firejail#4108] [Q&A] [answered]
The provided profile for Firefox includes `private-tmp`. Coming from other discussions ([#4103](https://github.com/netblue30/firejail/issues/4103) and [#3146](https://github.com/netblue30/firejail/issues/3146)), my approach would include multiple active instances of Firefox, with `--ignore=private-tmp`.

I could not find a reason for the enabled `private-tmp` option for Firefox. `/tmp/` is generally seen as world-writeable, so I don't see much of a security concern with Firefox having access to `/tmp/`.

However, I do not want to establish a setup with possible security concerns, thus the question:

What was the reason to include `private-tmp` in the default Firefox profile (`/etc/firejail/firefox-common.profile`) and what implications would `--ignore=private-tmp` entail?

**Environment**
 - Ubuntu 20.04.2 LTS
 - Firefox 86.0 (from Ubuntu focal repository)
 - firejail version 0.9.62

Compile time support:
        - AppArmor support is enabled
        - AppImage support is enabled
        - chroot support is enabled
        - file and directory whitelisting support is enabled
        - file transfer support is enabled
        - firetunnel support is enabled
        - networking support is enabled
        - overlayfs support is enabled
        - private-home support is enabled
        - seccomp-bpf support is enabled
        - user namespace support is enabled
        - X11 sandboxing support is enabled

**Comment by rusty-snake** [comment, marked as answer]
It's safe to `ignore private-tmp` if you use `whitelist /tmp/foo`:

```
ignore private-tmp
whitelist /tmp/.X11-unix
whitelist /tmp/firefox
```

> I could not find a reason for the enabled `private-tmp` option for Firefox.
> What was the reason to include `private-tmp` in the default Firefox profile

You will hardly never find a reason why restriction foo is in profile bar. `private-tmp` isolates `/tmp` inside the sandbox from `/tmp` of the system. And isolation between your system and the sandbox is that what you want if you use firejail.

> `/tmp/` is generally seen as world-writeable, so I don't see much of a security concern with Firefox having access to `/tmp/`.

Replace tmp with D-Bus:
&nbsp;&nbsp; D-Bus is generally seen as interprocess-communication, so I don't see much of a security concern with Firefox having access to it.

Some programs may have sockets in `/tmp`, store files in `/tmp` which are executed later (e.g. `cargo install` IIRC) and there are
 - [CWE-377: Insecure Temporary File](https://cwe.mitre.org/data/definitions/377.html)
 - [CWE-378: Creation of Temporary File With Insecure Permissions](https://cwe.mitre.org/data/definitions/378.html)
 - [CWE-379: Creation of Temporary File in Directory with Insecure Permissions](https://cwe.mitre.org/data/definitions/379.html)

> what implications would `--ignore=private-tmp` entail?

Access to `/tmp` might open another sandbox escape vector.

**Reply by BreakfastSerial** [reply]
Thank you very much for the elaborate answer! I appreciate it!

**Reply by BreakfastSerial** [reply]
A follow-up question, closely related.

If `private-tmp` and `whitelist=/tmp/foo` works, how come `private-home` does not work with `~/foo`?

**Reply by rusty-snake** [reply]
`private-tmp` is just syntasic sugar for:
```
whitelist /tmp/<Xauth>
whitelist /tmp/.X11-unix
read-only /tmp/.X11-unix
whitelist /tmp/pulse-*
```

Historical it was required because `whitelist` had no globing support.

It's implementation just adds such `whitelist` commands:
https://github.com/netblue30/firejail/blob/09673cd30deb8e2df0bb662a64f0b610b24144c8/src/firejail/fs.c#L1222-L1269

While `private-home` is a complete other command. (Except for `private-{bin,etc,opt,srv}` all `private-*` share not really more then the prefix).

**Reply by BreakfastSerial** [reply]
I appreciate the detailed answer and swift response. Thank you very much!
